lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1926,8 +1928,46 @@ internal class EvaluatingCompiler(
         }
     }
 
-    private fun compileExec(node: ExprNode): ThunkEnv {
-        TODO()
+    private fun compileExec(node: Exec): ThunkEnv {
+        val (procedureName, args, metas: MetaContainer) = node
+        val procedure = procedures[procedureName.name] ?: err(
+            "No such stored procedure: ${procedureName.name}",
+            ErrorCode.EVALUATOR_NO_SUCH_PROCEDURE,
+            errorContextFrom(metas).also {
+                it[Property.PROCEDURE_NAME] = procedureName.name
+            },
+            internal = false)
+
+        // Check arity
+        if (args.size !in procedure.signature.arity) {
```

Comment:
Just found something weird regarding `IntRange`: `IntRange(2,1)` is valid, no exception, but internally it assumes start <= end ([link](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/-int-range/)). `1 in IntRange(1,2)=true` but `1 in IntRange(2,1)=false`. Not to say we should do anything about it but maybe good to know.

Reply:
Hmm interesting. I don't think that internally `IntRange` assumes start <= end. The Kotlin 1.3 jar even defines an `EMPTY` range as `IntRange(1, 0)`. It seems `in` (i.e. `contains`) only returns true if and only if `start <= value && value <= last`, so the behavior makes some sense.

This leads me to think that nothing should be done unless we decide not to use `IntRange`.